web/bitwarden/organization.go

```diff
@@ -1,64 +1,93 @@
-	}
-	orgKey, err := setting.OrganizationKey()
-	if err != nil {
-		inst.Logger().WithField("nspace", "bitwarden").
-			Infof("Cannot read the organization key: %s", err)
-		return nil, err
+func (r *organizationRequest) toOrganizationAndCollection(
+	inst *instance.Instance,
+) (*bitwarden.Organization, *bitwarden.Collection) {
+	email := inst.PassphraseSalt()
+	o := bitwarden.Organization{
+		Name: r.Name,
+		Members: map[string]bitwarden.OrgMember{
+			inst.Domain: {
+				Email:  string(email),
+				Key:    r.Key,
+				Status: bitwarden.OrgMemberConfirmed,
+			},
+		},
 	}
-	key, err := crypto.EncryptWithRSA(setting.PublicKey, orgKey)
-	if err != nil {
-		inst.Logger().WithField("nspace", "bitwarden").
-			Infof("Cannot encrypt with RSA: %s", err)
-		return nil, err
+	c := bitwarden.Collection{
+		Name: r.CollectionName,
 	}
+	md := metadata.New()
+	md.DocTypeVersion = bitwarden.DocTypeVersion
+	o.Metadata = *md
+	c.Metadata = *md
+	return &o, &c
+}
 
-	email := inst.PassphraseSalt()
+// https://github.com/bitwarden/jslib/blob/master/common/src/models/response/profileOrganizationResponse.ts
+type organizationResponse struct {
+	ID             string  `json:"Id"`
+	Identifier     *string `json:"Identifier"`
+	Name           string  `json:"Name"`
+	Key            string  `json:"Key"`
+	Email          string  `json:"BillingEmail"`
+	Plan           string  `json:"Plan"`
+	PlanType       int     `json:"PlanType"`
+	Seats          int     `json:"Seats"`
+	MaxCollections int     `json:"MaxCollections"`
+	MaxStorage     int     `json:"MaxStorageGb"`
+	SelfHost       bool    `json:"SelfHost"`
+	Use2fa         bool    `json:"Use2fa"`
+	UseDirectory   bool    `json:"UseDirectory"`
+	UseEvents      bool    `json:"UseEvents"`
+	UseGroups      bool    `json:"UseGroups"`
+	UseTotp        bool    `json:"UseTotp"`
+	UseAPI         bool    `json:"UseApi"`
+	UsePolicies    bool    `json:"UsePolicies"`
+	UseSSO         bool    `json:"UseSSO"`
+	UseResetPass   bool    `json:"UseResetPassword"`
+	HasKeys        bool    `json:"HasPublicAndPrivateKeys"`
+	ResetPass      bool    `json:"ResetPasswordEnrolled"`
+	Premium        bool    `json:"UsersGetPremium"`
+	Enabled        bool    `json:"Enabled"`
+	Status         int     `json:"Status"`
+	Type           int     `json:"Type"`
+	Object         string  `json:"Object"`
+}
+
+func newOrganizationResponse(inst *instance.Instance, org *bitwarden.Organization) *organizationResponse {
+	m := org.Members[inst.Domain]
 	return &organizationResponse{
-		ID:             setting.OrganizationID,
-		Name:           consts.BitwardenCozyOrganizationName,
-		Key:            key,
-		Email:          string(email),
+		ID:             org.ID(),
+		Identifier:     nil, // Not supported by us
+		Name:           org.Name,
+		Key:            m.Key,
+		Email:          m.Email,
 		Plan:           "TeamsAnnually",
-		PlanType:       5, // TeamsAnnually plan
-		Seats:          2,
+		PlanType:       9,  // TeamsAnnually plan
```

Comment:
This number may be subject to future changes. Shouldn't we recreate [this enum](https://github.com/bitwarden/jslib/blob/1016bbfb9eb28c220de8d2ab86d1f2757328f254/common/src/enums/planType.ts) in go or maybe use a constant?

Reply:
No, we don't really care. It's like `Seats`: the client is expecting a value, but it doesn't do anything from it. It is normally checked server-side.